website/blog/2020-10-1-graphback-datasync.md

```diff
@@ -0,0 +1,148 @@
+---
+
+GraphQL subscriptions are a perfect fit for situations where users like to receive instant updates for some specific types of data. Subscriptions are very flexible and allow developers to connect directly to the stream of the changes generated on the server side.
+Subscriptions deliver delta updates to the client only when clients are subscribed. To receive partial updates after an application is restarted developers can use the concept of delta queries.
+## Delta queries
+Delta queries are a concept that allow users to perform a query that will return partial data based on a specific separator that is usually represented as a sequence or timestamp or any date series that will separate a sorted dataset into data we have seen and a delta that should be returned to clients.
+
+![](../static/img/syncdiagram.png)
+
+## Delete is challenging
+Not every dataset can support delta queries. Every deletion of the data will remove it from the database which means that sorting and splitting the dataset to retrieve updates will destroy it.
+
+## Conflict resolution
+In an offline-first application, where a client may often switch between being online and offline, conflicts between client-side and server-side data are almost guaranteed to occur. It is of utmost importance these conflicts are detected and some mechanism is  in place to resolve these conflicts.
+
+![](../static/img/conflictdiagram.png)
+
+### Solution?
+
+![](https://i.memeful.com/memes/R8nXE1b/Matrix-Morpheus.jpg)
+
+- Stop deleting data in the table to introduce soft deletes 
+- Have separate table for delete operations and merging results when sending data to clients
+- Have separate table that can be used to query diffs 
+
+These concepts are very common and are often needed when building applications that will consume large amount of data that needs to be available on the client - especially when the client is designed to work without network connectivity.
+
+### GraphQL CRUD spec for delta queries
+
+That is why we have built and included delta operations as well as conflict resolution into GraphQLCRUD.org specification.
+Today we are glad to announce that DeltaQueries as well as Server-side Conflict Resolution are officially available as part of the Graphback.dev project that implements this specification.
+Graphback is a library that lets you auto-generate your GraphQL Data layer at runtime. The latest release of Graphback provides support for delta queries as well as server-side conflict resolution.
+
+## What is DataSync?
+
+DataSync is short for data synchronization and that’s what we set out to do with this package. When a GraphQL client goes offline for sometime, e.g. because of a poor network connection, the data cached by the client can get outdated pretty fast and without a live subscription to fetch those changes it becomes imperative that a client provide some way to be up to date with the data as soon as it comes back online. That is DataSync.
```

Comment:
I see "this package" mentioned but no link or reference to `@graphback/datasync`.